### DISCUSSION TITLE
Are there any tips for making the compiler run faster?

### DISCUSSION THREAD

**Discussion by JasoonS** [foundry-rs/foundry#3525] [General]
I know foundry runs faster than other competitors for pure solidity. The issue is that with foundry (forge) you end up writing a ton more solidity; for us for every line of production solidity code we have probably 15x as many lines for test, scripts, deployments etc.

I've tried making all my contracts use interface files when referencing other contracts. But all the tests require the actual file to deploy - no now every change I make to a production contract has to compile over ~40 other contracts - and I don't see a way around that.

Even on a single file solidity contract test the DX is quite disruptive to have to wait for the compilation each time (compared to hardhat tests for example).

Two questions of discussion:
1. Are there heuristic of kinds of things to watch out for to keep the compiler fast? How does inheritance affect the compiler speed? Does use of libraries slow down compilation? etc
2. Could forge benefit from a type of compilation filtering? So that only 1 test contract re-compiles and the rest is ignored so that the DX can be smoother with faster feedback when working on the contract?

**Comment by moose-code** [comment]
Super interested in this. Loving foundry but currently waiting on the order of ~70s for compilation after contract changes. Thanks in advance for any tips!

**Comment by mattsse** [comment]
Does this happen with `--via-ir` enabled? ref https://github.com/foundry-rs/forge-std/issues/225

**Reply by JasoonS** [reply]
No, it wasn't with`--via -ir` (the contracts actually run into some issues if we try that flag).

**Reply by JasoonS** [reply]
I'll share the contracts publically in the next 2 weeks.
